aksel.nav.no/website/pages/templates/skjemavalidering/demo.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect } from "react";
+import { ArrowLeftIcon, ArrowRightIcon } from "@navikt/aksel-icons";
+import {
+  Button,
+  ErrorSummary,
+  HGrid,
+  Heading,
+  Page,
+  Radio,
+  RadioGroup,
+  TextField,
+  VStack,
+} from "@navikt/ds-react";
+
+const validatePersonnummer = (p: string) => {
+  if (p.length === 0) {
+    return "Du må fylle ut personnummer.";
+  }
+  if (!/^\d{11}$/.test(p)) {
+    return "Personnummer må være 11 siffer.";
+  }
+  return "";
+};
+
+const Example = () => {
+  const errorSummaryRef = React.useRef<HTMLDivElement>(null);
+
+  const [hasTriedToSubmit, setHasTriedToSubmit] = React.useState(0);
```

Comment:
Might be more of a preference thing, but i would group these two collections of states into two states, so something like:
```tsx
const [formState, setFormState] = useState({
 submitted: false,
 retries: 0  
})

const [formValues, setFormValue] = useState({
 personnummer: {
    value: "",
    error: ""
  },
...
```

Could even take it further to use a single state where formstates and values are separated by a object key. But might make updating values more cumbersome.

Reply:
I went with separate objects for values and errors, otherwise it becomes quite cumbersome.